### PR TITLE
Add developer email to env vars

### PR DESCRIPTION
**Pull request by lexer21**
I have added developer email to the env variables so when getting the access token the "correct" email will be used in the user agent.

### CODE REVIEW

File: crates/stash-api/src/sync/fetcher.rs [hunk 1]
```diff
@@ -63,6 +63,7 @@ pub(crate) fn start_fetcher(
     scheduler_tx: Sender<SchedulerMessage>,
 ) -> std::thread::JoinHandle<()> {
     std::thread::spawn(move || {
+        // Should this be prefixed with POE_ or is unused currently?
         let client_id = std::env::var("CLIENT_ID").unwrap();
```
Comment (maximumstock) [inline crates/stash-api/src/sync/fetcher.rs:67]: Good question! `stash-api/src/sync` was the first, sync version I built based on threads as an exercise but I simply have not deleted it yet. This whole module should essentially just be removed.

**Comment by maximumstock** [comment]
Wonderful, thank you!

**Comment by lexer21** [comment]
@maximumstock I have added the email (mxmlnstock@gmail.com) env var to the ci test pipeline, so it should pass the tests now. However I would like to add that it isn't good idea to rely on external api's while running unit tests, i think we should mock them instead.

**Comment by maximumstock** [comment]
> @maximumstock I have added the email ([mxmlnstock@gmail.com](mailto:mxmlnstock@gmail.com)) env var to the ci test pipeline, so it should pass the tests now. However I would like to add that it isn't good idea to rely on external api's while running unit tests, i think we should mock them instead.

Thanks for your help! I know its a bit finicky right now to contribute but I appreciate a ton :). I built upon it here https://github.com/maximumstock/poe-stash-indexer/commit/1fe8b6a5e8f76ddd6c1e3bd846a8f010d690dc35 and added it throughout.

What do you think would be the benefits of mocking all this? The only thing I see at the moment is that for those couple of days, where the stash tab API is offline in between challenge leagues, we don't get anything passing through CI without intervening. But in turn the integration test gives me quick and easy feedback on when something is broken (given I regularly get PRs via dependabot), even if "broken" means that the API is down. I see where you are coming from and agree that my current approach is not pure in a software testing sense or considered best practice, but it seems like a practical solution so far. 

As for testing in general, the pipeline in `stash-api` can be broken down further and then the response handling could be unit-tested more nicely.

**Pull request merged**